File: 1_iterations/binary_gap.py
def solution(N):
    max_len = 0
    length = 0
    remainder = 0
    while remainder != 1:
        remainder = N%2
        N //=2
    while N>0:
        remainder = N%2
        N //= 2
        if remainder == 1:
            max_len = max(max_len, length)
            length = 0
        else:
            length += 1
    return max_len

File: 1_iterations/binary_gap.cpp
#include <algorithm>

using namespace std;

int solution(int N) {
  int ret = 0;
  int cnt = 0;

  while (N%2 == 0) N /= 2;

  while (N>0) {
    if (N%2 == 1) {
      ret = max(ret, cnt);
      cnt = 0;
    }
    else cnt++;
    N /= 2;
  }
  return ret;
}
